src/overlay/Peer.cpp

```diff
@@ -957,15 +976,35 @@ Peer::sendSendMore(uint32_t numMessages)
     sendMessage(msgPtr);
 }
 
+void
+Peer::sendSendMore(uint32_t numMessages, uint32_t numBytes)
+{
+    ZoneScoped;
+    StellarMessage m;
+    m.type(SEND_MORE_EXTENDED);
+    m.sendMoreExtendedMessage().numMessages = numMessages;
+    m.sendMoreExtendedMessage().numBytes = numBytes;
+
+    auto msgPtr = std::make_shared<StellarMessage const>(m);
+    sendMessage(msgPtr);
+}
+
 void
 Peer::recvSendMore(StellarMessage const& msg)
 {
     ZoneScoped;
-    releaseAssert(msg.sendMoreMessage().numMessages > 0);
+    releaseAssert(getNumMessages(msg) > 0);
+    if (mFlowControlBytes)
+    {
+        releaseAssert(msg.type() == SEND_MORE_EXTENDED);
```

Comment:
this is wrong: if a peer sends us the wrong message type/bad data, we should not crash but drop it instead

Reply:
also the opposite is true (missing else): we should not allow a peer to send us `SEND_MORE_EXTENDED` if we didn't enable flow control in bytes

Reply:
Good catch, both cases are handled in `Peer::recvMessage(StellarMessage const& stellarMsg)` now where we perform other validity checks, so this assert can stay. Added test coverage as well.